frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -73,6 +75,54 @@ type ConvertOpt struct {
 }
 
 func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State, *Image, *binfotypes.BuildInfo, error) {
+	ds, bi, err := toDispatchState(ctx, dt, opt)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return &ds.state, &ds.image, bi, nil
+}
+
+func Dockefile2Outline(ctx context.Context, dt []byte, opt ConvertOpt) (*outline.Outline, error) {
+	ds, _, err := toDispatchState(ctx, dt, opt)
+	if err != nil {
+		return nil, err
+	}
+	o := ds.Outline(dt)
+	return &o, nil
+}
+
+func ListTargets(ctx context.Context, dt []byte) (*targets.List, error) {
+	dockerfile, err := parser.Parse(bytes.NewReader(dt))
+	if err != nil {
+		return nil, err
+	}
+	stages, _, err := instructions.Parse(dockerfile.AST)
+	if err != nil {
+		return nil, err
+	}
+
+	l := &targets.List{
+		Sources: [][]byte{dt},
+	}
+
+	for i, s := range stages {
+		if s.Name == "" && i != len(stages)-1 {
```

Comment:
I'm curious why unnamed stages are skipped.
Could it use the stage number instead of the name?

Reply:
We do not allow numbers for `--target` (and I don't think we should as it is not a good solution) so there isn't much to do with these rows other than noise. Notice that the last stage does not need to be named because that can be built without a name as well.

I think we should also promote to users that it is a good idea to name your stages(unless completely internal).

Reply:
Could we maybe leave the `Name` empty and include the remaining metadata anyways?

I agree it doesn't make sense for `--target`, but users might be surprised to see an entire stage from their Dockerfile missing.

Reply:
> but users might be surprised to see an entire stage from their Dockerfile missing.

What would they do with such a stage anyway in the Dockerfile? It looks like an invalid use case.

Reply:
For me, I'm collecting all the image sources in the dockerfile.
It's more about data collection/analysis than actually using the stage.

Reply:
I have removed this filtering